### ISSUE TITLE
Even more bugs.

### ISSUE THREAD

**Issue by Akitori** [AbrarSyed/SecretRoomsMod-forge#201]
Hiya. So i thought i nailed down every single bug and thing since nothing gave me issues, until i decided to try the secret lever. 

"A fatal error has occurred, this connection is terminated." That occurs every single time with the secret lever, and nothing console or client log wise tells me WHY.  

Other bugs I noticed. Trapdoors don't exactly work. Sure, the block looks closed when powered, but it doesn't allow me to interact with the empty air space as expected.

Iron Trapdoor is renamed to wooden trap door for some reason.

Secret redstone sometimes turns transparent/is no longer recognized as a block at times. At least, client side.

Going to keep testing and see if i find anything.

**Comment by Wyn-Price** [comment]
ok ill look into these issues. I tested everything before launch, but my guess is your using a server which could cause issues.

**Comment by Wyn-Price** [comment]
also what were you disguising the blocks as ?

**Comment by Akitori** [comment]
I actually have a video showing everything, I shall send it to you once its rendered

**Comment by Wyn-Price** [comment]
ok thanks

**Comment by Akitori** [comment]
oh boy looking at this latest log makes me worried. 

https://youtu.be/HJzPMVvcsWs

also here's my latest log. 
[latest.log](https://github.com/AbrarSyed/SecretRoomsMod-forge/files/1492064/latest.log)

**Comment by Wyn-Price** [comment]
thanks! ill look into whats going on right now

**Comment by Wyn-Price** [comment]
ok i fixed the first issue with the pressure plates

**Comment by Wyn-Price** [comment]
with trapdoors, their meant to be like that. 
![trapdoor](https://user-images.githubusercontent.com/15876682/33084625-54683e36-ceda-11e7-8385-096395e35497.gif)

**Comment by Wyn-Price** [comment]
ive fixed the issue with the lang

**Comment by Wyn-Price** [comment]
fixed the issue with the collision boxes of the trap doors

**Comment by Wyn-Price** [comment]
The reason there were so many bugs is I basically rewrote the mod to make it run smother and so fences and things would connect together. I did some bug testing but obviously not enough

**Comment by Wyn-Price** [comment]
I have no idea why theres colors. Not my mod so i suggest looking through all your mods and see one of the does that ?

**Comment by Akitori** [comment]
Its another mod, found out from testing.

**Comment by Wyn-Price** [comment]
I dont understand your issue with the door, it works for me on a server. 
![ezgif com-optimize 3](https://user-images.githubusercontent.com/15876682/33087252-f24a6a6e-cee1-11e7-8693-878abd65c4fe.gif)

**Comment by Akitori** [comment]
wonder if something else is interfering then. I'll try some testing then, see if another mod is interfering.

**Comment by Wyn-Price** [comment]
could you try it with the mod on its own. As well as the light detector, that also seems to be working fine on my end

**Comment by Akitori** [comment]
yeah I'll see what's going on. Hopefully the lever bug is fixed though.

**Comment by Wyn-Price** [comment]
What was the problem with the lever, Ive tried it and i cant seem to find anyway to make it crash

**Comment by Akitori** [comment]
secret lever caused crash when right clicked. Might be something interfering.

**Comment by Wyn-Price** [comment]
ok i think i figured out why your getting alot of diffrent bugs than me. It seems like you have a mod that is changing how block updated work and for some reason this causes it to crash. I might be wrong but it would explain the door and the light detector.

Could you send the log from the lever crash

**Comment by Akitori** [comment]
Right, i am back from secret rooms solely alone on a server.

The bug still persists. I can upload a video for you if you wish. 

The server's version is 2509, i have no clue if that makes a difference.

**Comment by Akitori** [comment]
https://youtu.be/oe-ur_kLp8I

logs from this video.

[latest.log](https://github.com/AbrarSyed/SecretRoomsMod-forge/files/1492423/latest.log)

**Comment by Wyn-Price** [comment]
right ok. I think i have an idea of whats going on but im looking more into it

**Comment by Wyn-Price** [comment]
Does the bugs happen in single player ?

**Comment by Akitori** [comment]
haven't tested, will do now.

**Comment by Wyn-Price** [comment]
And in the server, could you send a screenshot of the f3 screen while looking at a unpowered secret door, then don the same with the door being powered

**Comment by Wyn-Price** [comment]
could you send me the fml-client-latest.log and fml-server-latest.log for the lever crash

**Comment by Akitori** [comment]
lever crash leaves literally nothing behind in logs. I don't know why. 
unpowered
![2017-11-21_12 20 23](https://user-images.githubusercontent.com/24502131/33089563-960094a2-ceb6-11e7-9b5a-d21106b2d7a0.png)

powered
![2017-11-21_12 20 32](https://user-images.githubusercontent.com/24502131/33089575-9f17517a-ceb6-11e7-86c1-287ba97b504a.png)

and logs despite nothing being in there

[fml-server-latest.log](https://github.com/AbrarSyed/SecretRoomsMod-forge/files/1492517/fml-server-latest.log)

[fml-client-latest.log](https://github.com/AbrarSyed/SecretRoomsMod-forge/files/1492518/fml-client-latest.log)

**Comment by Wyn-Price** [comment]
ok thats very intresting. I think i understand whats happening

**Comment by Akitori** [comment]
err I noticed. Why isn't the state being set to powered?

**Comment by Wyn-Price** [comment]
im not sure. Ive just managed to replicate the issue tho, so ill be able to most defnialy fix these issues. Its 
 a serverside issue btw

**Comment by Akitori** [comment]
oh alright. Glad i could help!

**Comment by Wyn-Price** [comment]
for some reason, when the door is updated, the update packet is sending to the client with a blockstate of null.

**Comment by Wyn-Price** [comment]
um i think i found the problem
![image](https://user-images.githubusercontent.com/15876682/33090707-d84caa22-ceec-11e7-9922-7f9706ba894c.png)

**Comment by Akitori** [comment]
Oh no. That uh....Doesn't seem good.

**Comment by Wyn-Price** [comment]
oh my god

**Comment by Wyn-Price** [comment]
ok i fixed it

**Comment by Akitori** [comment]
what was the problem?

**Comment by Wyn-Price** [comment]
The only thing I havent managed to replicate is the lever.

**Comment by Akitori** [comment]
so it should be fixed in the next update then? would that also fix the weird lever bug/daylight bug?

**Comment by Akitori** [comment]
Yeah lever is a mystery....It says server error, kicks me, but I'm able to rejoin. Seems like a server based error rather than client

**Comment by Wyn-Price** [comment]
could you send the log for the server kicking you and the log from the client of you being kicked

**Comment by Akitori** [comment]
I'll try seeing if i get anything.

**Comment by Wyn-Price** [comment]
aight thanks

**Comment by Wyn-Price** [comment]
Ah ok i get it now. The issue is same issue as the door. but a little diffrent. The method readfromNBT's name is being changed, but not in ISecretTileEntity, which ends up that its trying to call a method that dosnt exist

**Comment by Akitori** [comment]
found a uh...issue? It kinda works?

to explain further.:

![2017-11-21_13 58 57](https://user-images.githubusercontent.com/24502131/33093977-2e76e404-cec4-11e7-910e-926b5fb3f2c3.png)
![2017-11-21_13 59 00](https://user-images.githubusercontent.com/24502131/33093988-30656484-cec4-11e7-923c-47a7b7e863b5.png)

good news is that the lever isn't broken anymore!

**Comment by Akitori** [comment]
this bug only happens on iron doors (also why is there no sound when redstone powered?)

**Comment by Wyn-Price** [comment]
does it happen when right click or when powered with redstone

**Comment by Wyn-Price** [comment]
does the redstone behavoir differ from when you click it with your hand

**Comment by Akitori** [comment]
well with the wooden door it works fine. Iron door is the only one with issues and i can't exactly right click a iron door open.

**Comment by Wyn-Price** [comment]
ah yes of corse. Ill look into it now

**Comment by Akitori** [comment]
oh and a crash with placing secret redstone...ok.

[crash-2017-11-21_14.04.22-client.txt](https://github.com/AbrarSyed/SecretRoomsMod-forge/files/1492813/crash-2017-11-21_14.04.22-client.txt)

**Comment by Akitori** [comment]
i placed a secret redstone on top of a secret redstone and that happened. idk why tho

**Comment by Wyn-Price** [comment]
could you try that again, does it do the same thing

**Comment by Akitori** [comment]
Yes, did it three times, three crashes.

**Comment by Akitori** [comment]
secret redstone on top of secret redstone = crash

**Comment by Wyn-Price** [comment]
right ok ill look into it now

**Comment by Wyn-Price** [comment]
ok i was able to replicate it.

**Comment by Wyn-Price** [comment]
ok i finally fixed it

**Comment by Akitori** [comment]
That's good to hear.

**Comment by Wyn-Price** [comment]
Ok finally finished the doors issues. Are you avaliable to make sure theres no bugs ?

**Comment by Akitori** [comment]
I'm available.

**Comment by Wyn-Price** [comment]
ah ok good heres the zip
[secretroomsmod-5.1.5.zip](https://github.com/AbrarSyed/SecretRoomsMod-forge/files/1493072/secretroomsmod-5.1.5.zip)

**Comment by Wyn-Price** [comment]
whats your curseforge username ?

**Comment by Akitori** [comment]
wew a crash from placing a torch lever on secret redstone. `i keep finding bugs send help`

[crash-2017-11-21_15.53.12-client.txt](https://github.com/AbrarSyed/SecretRoomsMod-forge/files/1493102/crash-2017-11-21_15.53.12-client.txt)

**Comment by Akitori** [comment]
Its WanderingSoulofTime

**Comment by Akitori** [comment]
actually its placing anything as i tried to place a default lever like i did for my gate bridge setup

[crash-2017-11-21_15.57.26-client.txt](https://github.com/AbrarSyed/SecretRoomsMod-forge/files/1493118/crash-2017-11-21_15.57.26-client.txt)

**Comment by Akitori** [comment]
huh interesting. Seems like powering the secret redstone then unpowering it causes it to keep its charge until a redstone update occurs nearby

**Comment by Wyn-Price** [comment]
with the crashes, what were you doing to make them occuer

**Comment by Wyn-Price** [comment]
wait i got it. Its when you try and place somthing on another secret block

**Comment by Akitori** [comment]
yeah, also the secret redstone is interesting...

not a bug exactly, but strange

https://www.youtube.com/watch?v=DfrPMHfHVzI

**Comment by Wyn-Price** [comment]
The redstone behaviour (bar the update thing) is normal. Try replacing the secret redstone with normal redstone to see if it has the same effect.

**Comment by Wyn-Price** [comment]
Ok ive nailed down the redstone updates issues

**Comment by Wyn-Price** [comment]
ok done

**Comment by Wyn-Price** [comment]
if you avaliable you can test it

**Comment by Wyn-Price** [comment]
[secretroomsmod-5.1.5.zip](https://github.com/AbrarSyed/SecretRoomsMod-forge/files/1493406/secretroomsmod-5.1.5.zip)

**Comment by Akitori** [comment]
I'm ready.

**Comment by Wyn-Price** [comment]
ok. Im gonna go take a break because ive been working for around 6 hours straight and im pretty tired but ill be back in around 30-60 minutes

**Comment by Akitori** [comment]
Hooray no issues so far to report. Looks good now! :D

**Comment by Wyn-Price** [comment]
Ok good. I'll release a version now

**Comment by Wyn-Price** [comment]
wait are you able to use daylight sensor

**Comment by Wyn-Price** [comment]
I just watched your video you were ok good. Thank you
